Client/Frontend/Browser/TabManager.swift

```diff
@@ -257,11 +257,13 @@ class TabManager : NSObject {
 
     // This method is duplicated to hide the flushToDisk option from consumers.
     func removeTab(tab: Browser) {
-        self.removeTab(tab, flushToDisk: true)
+        self.removeTab(tab, flushToDisk: true, notify: true)
         hideNetworkActivitySpinner()
     }
 
-    private func removeTab(tab: Browser, flushToDisk: Bool) {
+    /// - Parameter notify: if set to true, will call the delegate after the tab 
+    ///   is removed.
+    private func removeTab(tab: Browser, flushToDisk: Bool, notify: Bool) {
```

Comment:
To avoid having to change the method signatures for where removeTab is called, you can simplify this by adding a default parameter for notify. I would it to true so it matches the current behavior but opens up the method to passing in the notify flag in the case of closing private tabs.